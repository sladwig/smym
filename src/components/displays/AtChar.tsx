import React from 'react';
import { IUser } from '../../store/User';
import { Avatar } from '../User';
import classnames from 'classnames';

export const AtChar = ({ user, prev }: { user: IUser; prev: boolean }) => {
    return (
        <>
            {!user && <span className={classnames({ 'scale-in-center': prev !== !!user })}>@</span>}
            {user && (
                <span className={classnames({ 'scale-in-center': prev !== !!user })}>
                    <Avatar src={user.avatar} smaller={true} />
                </span>
            )}
        </>
    );
};
