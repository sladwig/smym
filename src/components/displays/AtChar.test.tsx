import { render } from '@testing-library/react';
import React from 'react';
import { AtChar } from './AtChar';

describe('AtChar', () => {
    test('renders', () => {
        const { container } = render(<AtChar />);
        expect(container).toMatchSnapshot();
    });
});
