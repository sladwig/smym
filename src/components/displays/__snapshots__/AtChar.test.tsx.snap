// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtChar renders 1`] = `
<div>
  <span
    class="scale-in-center"
  >
    @
  </span>
</div>
`;
